src/cms/previews/MarkdownPagePreview.tsx

```diff
@@ -1,18 +1,69 @@
 import { PreviewTemplateProps } from "netlify-cms-app";
 import React from "react";
+import { findDOMNode } from "react-dom";
 
+import {
+  DEFAULT_LOCALE,
+  LocaleCode,
+  LocaleDirection,
+  SUPPORTED_LOCALES,
+} from "@src/config/locales";
+import { LocaleContextProvider } from "@src/contexts/LocaleContext";
+import MainLayout from "@src/layouts/MainLayout";
+import { MuiLocaleProvider } from "@src/layouts/RootLayout";
 import { MarkdownTemplate } from "@src/templates/MarkdownPage";
 
-class MarkdownPagePreview extends React.Component<PreviewTemplateProps> {
+class MarkdownPagePreview extends React.PureComponent<PreviewTemplateProps> {
   public render() {
     const { entry, widgetFor } = this.props;
     return (
-      <MarkdownTemplate
-        title={entry.getIn(["data", "title"]) as string}
-        content={widgetFor("body")}
-      />
+      <LocaleContextProvider
+        initialLocale={this.locale}
+        path={this.path}
+        changeBodyDir={this.changeBodyDir}>
+        <MuiLocaleProvider>
+          <MainLayout>
+            <MarkdownTemplate
+              title={entry.getIn(["data", "title"])}
+              content={widgetFor("body")}
+            />
+          </MainLayout>
+        </MuiLocaleProvider>
+      </LocaleContextProvider>
     );
   }
+
+  private get locale(): LocaleCode {
+    const { entry } = this.props;
+    const slug = entry.get("slug");
+    if (!slug.includes(".")) {
+      return DEFAULT_LOCALE.code;
+    }
+
+    const splitRes = slug.split(".");
+    const locale = splitRes[splitRes.length - 1] as LocaleCode;
+    if (!SUPPORTED_LOCALES.includes(locale)) {
+      return DEFAULT_LOCALE.code;
+    }
+
+    return locale;
+  }
+
+  private get path(): string {
```

Comment:
I do not understand, why do we need a private getter here.

If we are using getter, the calling code will be `this.path()`

1. We agreed that we need a verb (or `is` keyword in case, when function returns boolean) in the function name to clearly determinate the use of the function. When I call `this.getPath()` I see from the function name what it does
1. If I see `this.path()` I will go to find this function by looking in constructor or by looking in the beginning of the class for a provate property.

Reply:
@goooseman You can't call getter as a function. See the usage above:
```jsx
<LocaleContextProvider path={this.path} {...}  />
```
It's a common React pattern to use getters when you need to pass computed string as a prop (not as a function call).

Reply:
@goooseman In JavaScript getters are accessed as object properties.  
And in React if you need to render some computed primitive that takes no arguments (like `path` or `locale` in the code above) it's common to create getter so you access them easily as `this.path` and `this.locale`.

Reply:
@goooseman It's like `const` and `let`. Whenever you sure that variable's value won't change - you define it with `const` keyword. Otherwise you defined it with `let` keyword.  
The same applies to getters. Whenever you sure method that returns primitive will take no arguments - define it as getter so it can be easily accessed as object property (`this.path`), otherwise defined it as method (`this.getComplexPath(someArgument)`).

Reply:
@alexeychikk nevertheless, let's use `this.getPath()` since it's slightly more obvious to the reader.